src/utils/cacheKeyFactory.ts

```diff
@@ -0,0 +1,112 @@
+import { useAccount } from 'wagmi'
+
+import type { uniqueTransactionIdentifierGenerator } from '@app/components/@molecules/TransactionDialogManager/stage/TransactionStageModal'
+import { useChainId } from '@app/hooks/useChainId'
+import type { TransactionItem } from '@app/transaction-flow/transaction'
+
+export const useQueryKeys = () => {
+  const { address } = useAccount()
+  const chainId = useChainId()
+
+  const globalKeys = [chainId, address]
+
+  return {
+    dogfood: (inputString?: string) => [...globalKeys, 'getAddr', inputString],
+    transactionStageModal: {
+      prepareTransaction: (
+        uniqueTransactionIdentifiers: ReturnType<typeof uniqueTransactionIdentifierGenerator>,
+      ) => [...globalKeys, 'prepareTx', uniqueTransactionIdentifiers],
+      transactionError: (transactionHash?: string) => [...globalKeys, 'txError', transactionHash],
+    },
+    claimDomain: (name: string, syncWarning: boolean) => [
+      ...globalKeys,
```

Comment:
this doesn't need global keys

Reply:
I'd prefer to just have the global keys actually be global

Reply:
changed my mind, put them under their own key